engine/engine.go

```diff
@@ -151,13 +154,16 @@ func (e *Engine) run(ctx context.Context) {
 	for {
 		select {
 		case err := <-e.consumer.Errors:
-			ErrHandler("", err)
+			ErrHandler(ctx, "received consumer error", err)
 		case msg := <-e.consumer.Messages:
-			e.handle(ctx, msg)
+			reqCtx, cancel := context.WithCancel(ctx)
+			defer cancel()
+			reqCtx = common.WithRequestId(reqCtx, msg.ID)
```

Comment:
I think @gedge is right that this cancel isn't being used or adding value so can be removed, but the `WithRequestId` should be kept.  So these lines could be replaced with:
```go
reqCtx = common.WithRequestId(ctx, msg.ID)
```